python-checks/src/main/java/org/sonar/python/checks/FlaskHardCodedSecret.java

```diff
@@ -84,9 +84,13 @@ private void verifyUpdateCallArgument(SubscriptionContext ctx, CallExpression ca
       .flatMap(TreeUtils.toOptionalInstanceOfMapper(RegularArgument.class))
       .map(RegularArgument::expression)
       .map(FlaskHardCodedSecret::getAssignedValue)
-      .filter(this::isIllegalDictArgument)
-      .ifPresent(expr -> ctx.addIssue(callExpression, String.format(MESSAGE, getSecretKeyType())));
+      .flatMap(this::getIllegalDictArgument)
+      .ifPresent(illegalArgument -> ctx.addIssue(illegalArgument, getMessage())
+        .secondary(callExpression.callee(), getMessage()));
```

Comment:
Having the primary message on the secondary location probably wouldn't make sense.
What about `The secret is used in this call.` as secondary message on the call site?